Automattic/woocommerce-payments
# Rename dispute action from `acceptTransactionDetailsDispute()` to `acceptDispute()`

Partially fixes https://github.com/Automattic/woocommerce-payments/issues/7363

#### Changes proposed in this Pull Request

- Rename dispute action from `acceptTransactionDetailsDispute()` to `acceptDispute()`.
- Go through `acceptDispute()` calls that got passed dispute id and change it so we pass dispute object because the new (renamed from `acceptTransactionDetailsDispute()`) `acceptDispute()` now accepts a dispute object instead of dispute id.
- Fix unit test due to the function rename.

#### Testing instructions

The action rename affects [useDispute()](https://github.com/Automattic/woocommerce-payments/blob/a4cded7f89e90f274468888a2878dbd9b927366d/client/data/disputes/hooks.ts#L26) hook and that hook is used in [client/disputes/details/index.tsx](https://github.com/Automattic/woocommerce-payments/blob/a4cded7f89e90f274468888a2878dbd9b927366d/client/disputes/details/index.tsx). This can be ignored because we're removing the legacy dispute details (part of https://github.com/Automattic/woocommerce-payments/issues/7363).

The rename also affects [client/disputes/redirect-to-transaction-details/index.tsx](https://github.com/Automattic/woocommerce-payments/blob/a4cded7f89e90f274468888a2878dbd9b927366d/client/disputes/redirect-to-transaction-details/index.tsx), so we need to test that redirection from dispute details page to transaction details page still works.

* Purchase a product with `4000000000000259` that will trigger a dispute.
* Go to Payments > Transactions > choose the transaction for your purchase.
* You will see a 'Submit evidence' button. Copy the link of that button.
* That link is for the dispute challenge page. Change it to open the legacy dispute details page.
```diff
- /wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdisputes%2Fchallenge&id=dp_abc
+ /wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdisputes%2Fdetails&id=dp_abc
```
* Expect to be redirected to the transaction page of the disputed payment.

Also
* Make sure unit tests are passing.

nb: as per the time of this writing, `merchant-orders-status-change.spec.js` test has been failing. Likely it's due to the change outside of this PR as I see that test fails in other PRs as well.

-------------------


**Post merge**
